redfish-core/lib/log_services.hpp

```diff
@@ -1475,6 +1528,229 @@ inline void requestRoutesJournalEventLogEntry(App& app)
             });
 }
 
+template <typename Callback>
+void getHiddenPropertyValue(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& entryId, Callback&& callback)
+{
+    auto respHandler = [callback{std::move(callback)},
+                        asyncResp](const boost::system::error_code ec,
+                                   std::variant<bool>& hiddenProperty) {
+        bool hiddenPropVal = false;
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "DBUS response error: " << ec;
+
+            // Error while retrieving Hidden Property. This can happen
+            // when the property is not found in the event log entry. In such
+            // cases try to process the entry as if it is a normal entry
+            // with Hidden property set to false
+            callback(hiddenPropVal);
+            return;
+        }
+        bool* hiddenPropValPtr = std::get_if<bool>(&hiddenProperty);
+        if (hiddenPropValPtr != nullptr)
```

Comment:
Same, can we just do 

         if (hiddenPropValPtr == nullptr)
         {
         
         internalError;
         return
         }

Reply:
Done